src/base.jl

```diff
@@ -2619,6 +2619,28 @@ function convert_component!(
     remove_component!(sys, old_load)
 end
 
+"""
+Set the number of a bus.
+"""
+function set_bus_number!(sys::System, bus::ACBus, number::Int)
+    if number in sys.bus_numbers
```

Comment:
Given https://github.com/NREL-Sienna/InfrastructureSystems.jl/issues/397, seems like `set_bus_number!` with the bus's current number should be a no-op